src/OpenFeature.Contrib.Providers.Flagd/README.md

```diff
@@ -107,3 +109,48 @@ var flagdProvider = new FlagdProvider(new Uri("http://localhost:8013"));
 var unixFlagdProvider = new FlagdProvider(new Uri("unix://socket.tmp"));  
 ```
 
+## In-process resolver type
+
+The flagd provider also supports the [in-process provider mode](https://flagd.dev/reference/specifications/in-process-providers/),
+which is activated by setting the `FLAGD_RESOLVER_TYPE` env var to `IN_PROCESS`.
+In this mode, the provider will connect to a service implementing the [flagd.sync.v1 interface](https://github.com/open-feature/flagd-schemas/blob/main/protobuf/flagd/sync/v1/sync.proto)
+and subscribe to a feature flag configuration determined by the `FLAGD_SOURCE_SELECTOR`.
+After an initial retrieval of the desired flag configuration, the in-process provider will keep the latest known state in memory,
+meaning that no requests need to be sent over the network for resolving flags that are part of the flag configuration.
+Updates to the flag configuration will be sent via the grpc event stream established between the in-process provider and
+the service implementing the `flagd.sync.v1` interface (e.g. [flagd-proxy](https://github.com/open-feature/flagd/tree/main/flagd-proxy)).
+
+Example of using the in-process provider mode:
+
+```csharp
+using OpenFeature.Contrib.Providers.Flagd;
+
+namespace OpenFeatureTestApp
+{
+    class Hello {
+        static void Main(string[] args) {
+
+            // set the host and port for flagd-proxy
+            Environment.SetEnvironmentVariable("FLAGD_HOST", "localhost");
+            Environment.SetEnvironmentVariable("FLAGD_PORT", "8015");
+            // set the resolver type to 'IN_PROCESS'
+            Environment.SetEnvironmentVariable("FLAGD_RESOLVER_TYPE", "IN_PROCESS");
+            // provide the flag source selector, e.g. the name of a Flags custom resource which is watched by the flagd-proxy
+            Environment.SetEnvironmentVariable("FLAGD_SOURCE_SELECTOR", "core.openfeature.dev/flags/sample-flags");
+
+            var flagdProvider = new FlagdProvider();
```

Comment:
Minor nit - example could be updated to use builder and config bound constructor. This then matches with the rest of provider documentations :)